datafusion/physical-expr/src/aggregate/first_last.rs

```diff
@@ -459,10 +493,52 @@ impl LastValueAccumulator {
     }
 
     // Updates state with the values in the given row.
-    fn update_with_new_row(&mut self, row: &[ScalarValue]) {
-        self.last = row[0].clone();
-        self.orderings = row[1..].to_vec();
-        self.is_set = true;
+    fn update_with_new_row(&mut self, row: &[ScalarValue]) -> Result<()> {
+        let value = &row[0];
+        let orderings = &row[1..];
+        // Update when
+        // - no value in the state
+        // - There is no specific requirement, but a new value (most recent entry in terms of execution)
+        // - There is a more recent entry in terms of requirement
+        if !self.is_set
+            || self.orderings.is_empty()
+            || compare_rows(
```

Comment:
I'm sure you are aware but https://docs.rs/arrow-row/latest/arrow_row/ will be a much faster way to perform row-based comparisons than relying on ScalarValue

Reply:
Indeed, however, here we are checking just a single row (row that have lowest value). Hence I don't think it is worth to conversion here.

Reply:
I agree that since it is a single column max comparison this is probably fine (and no worse than the current implementation). If  we need to optimize performance we could probably implement specialized implementations (like `FirstValue<ArrowPrimitiveType>` and skip the copying entirely.

That is likely a premature optimization at this point

Reply:
I re-reviewed and I agree that the `RowFormat` is not needed here (and in fact it may actually be slower) because, as @mustafasrepo  points out, this code uses `ScalarValue` to compare a single row per batch (it finds the largest/smallest row per batch using `lexsort_to_indices`). We would have to benchmark to be sure.